_drafts/floating-point-101.md

```diff
@@ -0,0 +1,444 @@
+        <div class="number-line__marker number-line__marker--8"></div>
+        <div class="number-line__marker number-line__marker--8"></div>
+        <div class="number-line__marker number-line__marker--8 number-line__marker--power"><span>2<sup>-1019</sup></span></div>
+    </div>
+    <figcaption><div class="recommendation recommendation--do"></div><i>fig. 3</i> — Do.</figcaption>
+</figure>
+
+> See how subnormal numbers are stored in binary by trying `8e-323` in [the calculator](#calculator).
+
+## The Enormous
+
+On the other hand, when the result of a calculation is so _large_ that it can't be represented – a situation called
+**overflow** – another special value is returned: infinity. It can be positive or negative, depending on the direction
+of overflow.
+
+Infinity obviously almost never is the correct result at all, but that's a feature, not a bug. Returning the maximum
+representable number would be much more unsafe – you'd end up with a _seemingly_ normal value, that would actually be
+off by 1, or perhaps by orders of magnitude, with no way to tell. Infinity makes it evident an overflow happened and the
+rules of calculations involving it are well-defined: basically, any such operation that doesn't produce a NaN (listed in
+[The Undefined](#the-undefined)) results in infinity (with sign rules applying, i.e. `Infinity * (-3) == -Infinity`).
+
+> See how infinity is stored in binary by trying `Infinity` in [the calculator](#calculator).
+
+Interestingly, one extra case where you get infinity is division by 0. Ordinarily in such case the answer is considered
+to be undefined, but IEEE 754 stipulates that the sign ought to be preserved, which is why NaN only is returned if the
+numerator is 0 too (0 divided by 0 – when there is no way to interpret the expression _at all_). Otherwise, the
+interpretation that's used is that the _limit_ of the expression is taken, approaching from either positive or negative
+numbers – here's where the sign of the zero uniquely plays a role. For instance: `123 / -0 == -Infinity`.
+
+## The More-or-Less
+
+In floating-point, what you see is usually not what you get. As outlined in [The Precision](#the-precision), bits don't
+quite grow on trees, so only a limited subset of points on the number line can be stored, and those points are all in
+base-2. These two facts combined are a source of constant friction between humans and computers, kept at bay thanks to
+an array of tricks.
+
+Whether it's a user providing data or you hard-coding values, the starting point for many real numbers is a string of
+characters representing the decimal value. We can, for instance, parse `"0.2"` as a double. Print that back and you get
+`0.2` as expected. That's not exactly what's stored though. If we calculate the value a bit more accurately based on the
+binary data, using [The Notation](#the-notation), we get `0.2000000000000000111022...`. That's evidently off! But just
```

Comment:
how exactly does the notation section help in parsing 0.2 as 0.2000001 ? 🤔 

Or, do you mean to say it's parsed as the closest possible binary, which is the above? worth clarifying!
(and I'd reuse the distance analogy lol to show a few possible points in binary, their value, and then us choosing the closest somehow)

Reply:
Ah, you go deeper right below.

I'd remove or put the 'but that's fine' line later on then, since it felt like you were skipping over this conversion which didn't make sense without the example below

Reply:
I see, went for the low-effort improvement and just removed "it's fine" :D